5minds/addict-ioc
# ♻️🐛💄 Overhaul tests

TL DR:
Make all tests great again.

**Changes:**

Tests:
1. Update all tests so they match the current implementation of the addict-ioc package.
2. Remove tests that cover functions which no longer exist.

Fixes:
1. Fix the "getKeysByTags" function: When retrieving registrations for multiple keys, a registration that has more than one matching key was returned multiple times (See #61).
2. Use `clone` for creating copies of an object, instead of `Object.assign` to prevent the original object references from being overwritten (See #62).

Enhancements:
1. Replace obsolete `node-uuid` package with `uuid`.
2. Add checks to `register` methods, which ensure that `key` is a string and `type` is not null or undefined (See  #58).

Miscellaneous:
1. Fix styling for all tests and .ts components.
2. Update the ESLint rules to match the ones we use in the ProcessEngine (we should consider updating the 5minds ESLint style).
3. Enhance Contributors info.
4. Use fixed dependency versions.
5. Remove `bluebird`, since its no longer used anywhere.

**Issues:**

Closes #53 
Closes #58
Closes #61 
Closes #62 

PR: #60

## How can others test the changes?

Run the tests.

## PR-Checklist

Please check the boxes in this list after submitting your PR:


## Example

<details>
<summary>
:warning: Before merging please click here to expand and follow the guide.
</summary>
<br>

Please use `:twisted_rightwards_arrows:` at the beginning of your merge commit title.

Example 1:

<pre>
<code>
:twisted_rightwards_arrows: :bug: Fix Wrong Text Decoration at ...
</code>
</pre>

To get your commit message, just copy the first part of this pull request.

Example 2:
<pre>
<code>
**Changes:**

- Fixes Wrong Text Decoration at ...
- Fixes some typos
- ...

**Issues:**

Closes #NumberOfFixedIssue

PR: #NumberOfThisPR
</code>
</pre>
</details>

## Emojis

<details>
<summary>
Expand for a list of most used Emojis.
</summary>
<br>

Please prefix your commit messages with an Emoji.

Ref: https://gitmoji.carloscuesta.me/

| Description              | Glyphe               | Emoji  |
|--------------------------|----------------------|--------|
| Bugfix                   | `:bug:`              | 🐛     |
| Configuration releated   | `:wrench:`           | 🔧     |
| Cosmetic                 | `:lipstick:`         | 💄     |
| Dependencies Downgrade   | `:arrow_down:`       | ⬇️     |
| Dependencies Upgrade     | `:arrow_up:`         | ⬆️     |
| Formatting               | `:art:`              | 🎨     |
| Improving Performance    | `:zap:`              | ⚡️     |
| Initial commit           | `:tada:`             | 🎉     |
| Linter                   | `:rotating_light:`   | 🚨     |
| Miscellaneous            | `:package:`          | 📦     |
| New Feature              | `:sparkles:`         | ✨     |
| Refactoring Code         | `:recycle:`          | ♻️     |
| Releasing / Version tags | `:bookmark:`         | 🔖     |
| Removing Stuff           | `:fire:`             | 🔥     |
| Tests                    | `:white_check_mark:` | ✅     |
| Work In Progress (WIP)   | `:construction:`     | 🚧     |

</details>
